sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -900,10 +889,41 @@ object JdbcUtils extends Logging {
       newTable: String,
       options: JDBCOptions): Unit = {
     val dialect = JdbcDialects.get(options.url)
+    executeStatement(conn, options, dialect.renameTable(oldTable, newTable))
+  }
+
+  /**
+   * Update a table from the JDBC database.
+   */
+  def alterTable(
+      conn: Connection,
+      tableName: String,
+      changes: Seq[TableChange],
+      options: JDBCOptions): Unit = {
+    val dialect = JdbcDialects.get(options.url)
+    conn.setAutoCommit(false)
     val statement = conn.createStatement
     try {
       statement.setQueryTimeout(options.queryTimeout)
-      statement.executeUpdate(dialect.renameTable(oldTable, newTable))
+      for (sql <- dialect.alterTable(tableName, changes)) {
+        statement.executeUpdate(sql)
+      }
+      conn.commit()
+    } catch {
+      case e: SQLException =>
+        if (conn != null) conn.rollback()
```

Comment:
I'm not sure if all JDBC servers support it. At least Spark thriftserver doesn't support it. How about we limit the scope to only support ALTER TABLE when `changes.length == 1`? Then we don't need to worry about atomic issues.

Reply:
I would check `changes.length == 1` only if `supportsTransactions()` is false.

Reply:
that also works, but may require more test cases (one more dimension to test)

Reply:
My tests have  ```changes.length == 1 (ALTER TABLE h2.test.alt_table ADD COLUMNS (C3 DOUBLE))``` and ```changes.length == 2 (ALTER TABLE h2.test.alt_table ADD COLUMNS (C1 INTEGER, C2 STRING))```